src/changefile/getChangedPackages.ts

```diff
@@ -3,11 +3,83 @@ import path from 'path';
 import minimatch from 'minimatch';
 import { ChangeFileInfo, ChangeInfoMultiple } from '../types/ChangeInfo';
 import { getChangePath } from '../paths';
-import { getChanges, getStagedChanges, git, fetchRemoteBranch, parseRemoteBranch } from 'workspace-tools';
+import { getChanges, getStagedChanges, git, fetchRemoteBranch, parseRemoteBranch, GitError } from 'workspace-tools';
 import { getScopedPackages } from '../monorepo/getScopedPackages';
 import { BeachballOptions } from '../types/BeachballOptions';
 import { PackageInfos, PackageInfo } from '../types/PackageInfo';
 
+/**
+ * Ensure that adequate history is available to check for changes between HEAD and `options.branch`.
+ * Otherwise attempting to get changes will fail with an error "no merge base".
+ */
+function ensureHistory(options: BeachballOptions) {
+  const { fetch, path: cwd, branch } = options;
+  const { remote, remoteBranch } = parseRemoteBranch(branch);
+
+  const fetchMitigationSteps = `- Omit the "--no-fetch" / "--fetch=false" option from the command line
+- Remove "fetch: false" from the beachball config
+- If this is a CI build, ensure that adequate history is being fetched
+  - For GitHub Actions (actions/checkout), add the option "fetch-depth: 0" in the checkout step`;
+
+  if (fetch) {
+    // Fetch the latest from the remote branch for comparison
+    console.log(`fetching latest from remotes "${remote}/${remoteBranch}"`);
+    fetchRemoteBranch(remote, remoteBranch, cwd);
+  } else {
+    // If fetching is disabled, ensure that the target branch is available for comparison locally
+    const hasTargetBranch = git(['rev-parse', '--verify', branch], { cwd }).success;
+    if (!hasTargetBranch) {
+      // This is most likely to happen in a CI build which does a shallow checkout (github actions/checkout
+      // does this by default) and for some reason also disables beachball fetching.
+      const mainError = `Target branch "${branch}" does not exist locally, and fetching is disabled.`;
+      console.error(`
+
+${mainError} Some possible fixes:
+- Fetch the branch manually: git fetch ${remote} ${remoteBranch}
+${fetchMitigationSteps}
+
+`);
+      throw new Error(mainError);
+    }
+  }
+
+  // Verify that HEAD and the target branch share history
+  const hasCommonCommit = git(['merge-base', branch, 'HEAD'], { cwd }).success;
+  if (!hasCommonCommit) {
+    // This might be a shallow repo, and it's necessary to unshallow the head branch for comparison
+    const isShallow = git(['rev-parse', '--is-shallow-repository'], { cwd }).stdout.trim() === 'true';
+    if (isShallow) {
+      if (fetch) {
+        // Fetch more history (if needed, this could be optimized later to only deepen by e.g. 100 commits at a time)
+        try {
+          console.log('This is a shallow clone. Unshallowing to check for changes...');
+          git(['fetch', '--unshallow'], { cwd, throwOnError: true });
```

Comment:
Can you confirm this "throwOnError" flag? I can't find doc for it: https://nodejs.org/api/child_process.html#child_processspawnsynccommand-args-options

Reply:
Ah it's added in https://github.com/microsoft/workspace-tools/pull/193 which isn't actually merged yet (I was testing this branch against a tgz of those changes at first)

Reply:
I got rid of the part that required the workspace-tools update